subworkflows/local/bam_variant_calling_single_vardictjava/main.nf

```diff
@@ -0,0 +1,72 @@
+//
+// Vardictjava germline calling
+//
+// For all modules here:
+// A when clause condition is defined in the conf/modules.config to determine if the module should be run
+
+include { VARDICTJAVA                                  } from '../../../modules/nf-core/vardictjava/main'
+include { GATK4_MERGEVCFS  as MERGE_VARDICTJAVA        } from '../../../modules/nf-core/gatk4/mergevcfs/main'
+include { SAMTOOLS_CONVERT as CRAM_TO_BAM              } from '../../../modules/nf-core/samtools/convert/main'
+
+workflow BAM_VARIANT_CALLING_SINGLE_VARDICTJAVA {
+    take:
+    cram          // channel: [mandatory] [ meta, cram, crai ]
+    dict          // channel: [optional]  [ meta, dict ]
+    fasta         // channel: [mandatory] [ fasta ]
+    fasta_fai     // channel: [mandatory] [ fasta_fai ]
+    intervals     // channel: [mandatory] [ interval.bed.gz, interval.bed.gz.tbi, num_intervals ] or [ [], [], 0 ] if no intervals
+
+    main:
+    versions = Channel.empty()
+
+    // Convert cram to bam
+    cram
+    .branch {meta, cram, crai ->
+        bam:    cram.extension == "bam"
+        cram:   cram.extension == "cram"}
+        .set{ch_bam_from_cram}
+
+    CRAM_TO_BAM(
+        ch_bam_from_cram.cram,
+        fasta,
+        fasta_fai
+    )
+
+    // Combine converted bam, bai and intervals
+    ch_bam_from_cram.bam
+        .mix(CRAM_TO_BAM.out.bam.join(CRAM_TO_BAM.out.bai, failOnDuplicate: true, failOnMismatch: true))
+        .combine(intervals)
+        .map{meta, bam, bai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], bam, bai, intervals ]}
+        .set{ ch_vardict_input}
```

Comment:
no cram support 😭 ?

Reply:
@maxulysse we should refactor this to only to cram -> bam once if tools like cnvkit and vardict are both selected.